File: Add Code Here/C++/leetcode/zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        string result="";
        if(numRows==1)
      return s;
        int step1,step2;
        int len=s.size();
        for(int i=0;i<numRows;++i){
            step1=(numRows-i-1)*2;
            step2=(i)*2;
            int pos=i;
            if(pos<len)
                result+=s.at(pos);
            while(1){
                pos+=step1;
                if(pos>=len)
                    break;
        if(step1)
          result+=s.at(pos);
                pos+=step2;
                if(pos>=len)
                    break;
        if(step2)
          result+=s.at(pos);
            }
        }
        return result;
    }
};

File: Add Code Here/JAVA/Java_Medium_Solutions/ZigZagConversion.java
class ZigZagConversion  {
    public String convert(String s, int numRows) {
        if (numRows == 1 || numRows >= s.length()) {
            return s;
        }
        StringBuilder[] strings = new StringBuilder[numRows];
        for(int i=0;i<numRows;i++)
        {
            strings[i] = new StringBuilder();
        }
        int j =0;
        int direction =1;
        for(int i=0;i<s.length();i++)
        {
            strings[j].append(s.charAt(i));
            if(j==0)
            {
                direction = 1;
            }
            if(j == numRows-1)
            {
                direction = -1;
            }
            j +=direction;
        }
        StringBuilder result = new StringBuilder();
        for (StringBuilder sb : strings) {
            result.append(sb);
        }

        return result.toString();
    }
}